### PR TITLE
Transform HLA classic RA and Dec values into HAP reference frame for better quality sourcelist comparisons

### PR DESCRIPTION
**Pull request by dulude**
**High-level summary**
* Integrated Rick White's read_hla_catalog.py code that converts the HLA Classic sourcelist RA and Dec values into the HAP reference frame. 
* Depending on the dataset, this may:
     1. eliminate a possible source of error in the RA and Dec comparisons when comparing HLA Classic sourcelists to their HAP counterparts
     2. improve the source matching when comparing HLA Classic sourcelists to their HAP counterparts
* See issue #544 

**Codebase changes**
* Added subdirectory drizzlepac/devutils/comparison_tools/read_hla
* Added subroutine correct_hla_classic_ra_dec() to drizzlepac/hapsequencer.py.  The _drizzlepac/hapsequencer.correct_hla_classic_ra_dec()_ subroutine interfaces with _drizzlepac/devutils/comparison_tools/read_hla/read_hla_catalog.read_hla_catalog()_ to perform the coordinate transform, and writes an updated version of the HLA classic (with the updated RA/Dec values) to the current working directory. If no failures are experienced in the conversion, this updated sourcelist is used by _compare_sourcelists.py_ to compare against the  corresponding HAP catalog.

### CODE REVIEW

**Comment by dulude** [comment]
> There is a problem in the compilation. Please see the results from the regression tests. Here is a snippet of the issue:
> 
> _ ERROR collecting drizzlepac/devutils/comparison_tools/read_hla/hla_test.py __
> ImportError while importing test module 'D:\a\1\s\drizzlepac\devutils\comparison_tools\read_hla\hla_test.py'.
> Hint: make sure your test modules/packages have valid Python names.
> Traceback:
> drizzlepac\devutils\comparison_tools\read_hla\hla_test.py:4: in
> from read_hla_catalog import read_hla_catalog
> drizzlepac\devutils\comparison_tools\read_hla\read_hla_catalog.py:6: in
> from . import omegaxyz
> E ImportError: attempted relative import with no known parent package

Hey @mdlpstsci : Issue resolved by commit [681e5a5](https://github.com/spacetelescope/drizzlepac/pull/548/commits/681e5a5327a2246d531aeef37140b88a234b2777)

File: drizzlepac/hapsequencer.py [hunk 1]
```diff
@@ -53,6 +55,67 @@
...
+                                                   verbose=True, trim=False, log_level=log_level)
+
+        # Identify RA and Dec column names in the new catalog table object
+        for ra_col_title in ["ra", "RA", "ALPHA_J2000", "alpha_j2000"]:
+            if ra_col_title in modcat.colnames:
+                true_ra_col_title = ra_col_title
+                log.debug("RA Col_name: {}".format(true_ra_col_title))
+                break
+        for dec_col_title in ["dec", "DEC", "Dec", "DELTA_J2000", "delta_j2000"]:
+            if dec_col_title in modcat.colnames:
```
Comment (mdlpstsci) [inline drizzlepac/hapsequencer.py]: Just a comment...suggest to convert strings to lower- or upper-case before comparison.

File: drizzlepac/devutils/comparison_tools/integrate_read_hla_catalog.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+import os
+import pdb
+import sys
+from astropy.table import Table
+from drizzlepac.devutils.comparison_tools import compare_sourcelists
+from drizzlepac.devutils.comparison_tools.read_hla import read_hla_catalog
+
```
Comment (mdlpstsci) [inline drizzlepac/devutils/comparison_tools/integrate_read_hla_catalog.py]: I am not sure what this routine is doing in this directory.  This code looks like it is fully integrated in hapsequencer.  If this is true, this file should be deleted.

File: drizzlepac/hapsequencer.py (hunk 1 above)
Comment (dulude) [reply drizzlepac/hapsequencer.py]: Resolved in commit [224e89f](https://github.com/spacetelescope/drizzlepac/pull/548/commits/224e89fe00f4431208dd872f35ec82023f9ca803)

File: drizzlepac/devutils/comparison_tools/integrate_read_hla_catalog.py (hunk 2 above)
Comment (dulude) [reply drizzlepac/devutils/comparison_tools/integrate_read_hla_catalog.py]: Yeah. this was just a test script I wrote to test the correct_hla_classic_ra_dec() subroutine without having to take the time to run a full runhapsequencer.py run. I'll delete it.

Comment (dulude) [reply drizzlepac/devutils/comparison_tools/integrate_read_hla_catalog.py]: Deleted in commit [e6e8d8e](https://github.com/spacetelescope/drizzlepac/pull/548/commits/e6e8d8e30f08eca61e3a2212e09708b5deca43e6). Issue resolved.

File: drizzlepac/devutils/comparison_tools/read_hla/read_hla_catalog.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+    # create a column name dictionary
+    d = dict([(x, 1) for x in cat.colnames])
+    for v in ["ra", "RA", "ALPHA_J2000", "alpha_j2000"]:
+        if v in d:
+            racol = v
+            break
+    else:
+        raise ValueError("RA column not found in", " ".join(cat.colnames))
+    for v in ["dec", "DEC", "Dec", "DELTA_J2000", "delta_j2000"]:
```
Comment (mdlpstsci) [inline drizzlepac/devutils/comparison_tools/read_hla/read_hla_catalog.py]: Same comment as previously stated in another routine (can be addressed later), convert comparisons all to lower- or upper-case.

Comment (dulude) [reply drizzlepac/devutils/comparison_tools/read_hla/read_hla_catalog.py]: Resolved by commit [9e55ef4](https://github.com/spacetelescope/drizzlepac/pull/548/commits/9e55ef4b4ae79063fdb1250143e3d7a2477372aa)

File: drizzlepac/devutils/comparison_tools/read_hla/1AAREADME [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+omegaxyz.py         Functions to access WCS correction database and apply them
+
+See the test code at the bottom of these modules for examples of how to use them.
+
+hla_test.py         Test script that creates both FITS images with and without
+                    corrections and also HLA catalogs with and without corrections.
+
+dods9.csh           Shell script that displays the test images in ds9 and overlays
+                    the catalogs with (green) and without (red) the WCS corrections
+                    as a test.
```
Comment (mdlpstsci) [inline drizzlepac/devutils/comparison_tools/read_hla/1AAREADME]: Update documentation here.

File: drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+from __future__ import print_function
+import omegaxyz, os
+from astropy.io import fits
+
+def read_hla_image(dataset, applyomega=True, verbose=False,
+        url="https://hla.stsci.edu/cgi-bin/getdata.cgi"):
+
+    """Return a astropy.io.fits HDUlist for the given HLA dataset.
+
```
Comment (mdlpstsci) [inline drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py]: Pep8 docstring - parameter description.

File: drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    if not os.path.exists(dataset):
+        dataloc = "{}?dataset={}".format(url,dataset)
+    else:
+        dataloc = dataset
+    if applyomega:
+        fh = omegaxyz.updatefits(dataloc, verbose=verbose)
+    else:
+        fh = fits.open(dataloc)
+    return fh
```
Comment (mdlpstsci) [inline drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py]: Same as above comment re: Pep8.

File: drizzlepac/devutils/comparison_tools/read_hla/read_hla_catalog.py (hunk 3 above)
Comment (dulude) [reply drizzlepac/devutils/comparison_tools/read_hla/read_hla_catalog.py]: It actually crashes with these changes. Above changes rolled back in commit [7e1e135](https://github.com/spacetelescope/drizzlepac/pull/548/commits/7e1e1350b491e9a3f7125f6a041aed0183547407)

File: drizzlepac/devutils/comparison_tools/read_hla/omegaxyz.py [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+
+    if not omega:
+        # parse the filename and extract the visit identification
+        dataset = getdataset(filename)
+        if not dataset:
+            return (crval, cdmatrix)
+        omega = getomegaxyz(dataset)
+
+    if omega[0] != 0 or omega[1] != 0 or omega[2] != 0:
+        d2r = math.pi/180
```
Comment (mdlpstsci) [inline drizzlepac/devutils/comparison_tools/read_hla/omegaxyz.py]: For later...should use functions in math or numpy to convert degrees -> radians and radians -> degrees.

File: drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py (hunk 5 above)
Comment (dulude) [reply drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py]: Issue resolved in commit [041e699](https://github.com/spacetelescope/drizzlepac/pull/548/commits/041e69974fe95fc578d00d35c63dc4bcb8eb88fa)

File: drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py (hunk 6 above)
Comment (dulude) [reply drizzlepac/devutils/comparison_tools/read_hla/read_hla_image.py]: Issue resolved in commit [c060957](https://github.com/spacetelescope/drizzlepac/pull/548/commits/c06095740dcc1e7aa3fef2c7a535cb2b03acbfc3)

File: drizzlepac/devutils/comparison_tools/read_hla/1AAREADME (hunk 4 above)
Comment (dulude) [reply drizzlepac/devutils/comparison_tools/read_hla/1AAREADME]: Issue resolved in commit [5d67d60](https://github.com/spacetelescope/drizzlepac/pull/548/commits/5d67d60f9d5c245fa1ba9514ce38324b87b1af5d)

File: drizzlepac/devutils/comparison_tools/read_hla/omegaxyz.py (hunk 7 above)
Comment (dulude) [reply drizzlepac/devutils/comparison_tools/read_hla/omegaxyz.py]: Issue resolved in commit [1f8da23](https://github.com/spacetelescope/drizzlepac/pull/548/commits/1f8da234a95cf22f0b6f15e51e55ff8e98425d89)

**Pull request merged**